File: Safari-611.1.10/LayoutTests/http/tests/multipart/multipart-html.php
<?php
header('Content-type: multipart/x-mixed-replace; boundary=boundary');
header('Connection: keep-alive');
echo "--boundary\r\n";
echo "Content-Type: text/html\r\n\r\n";
echo str_pad('', 5000);
?>

<script>
if (window.testRunner) {
    testRunner.dumpAsText();
    if (testRunner.setShouldDecideResponsePolicyAfterDelay)
        testRunner.setShouldDecideResponsePolicyAfterDelay(true);
}
</script>

<?php
for ($i = 0; $i <= 10; $i++) {
    echo "--boundary\r\n";
    echo "Content-Type: text/html\r\n\r\n";
    if ($i < 10) {
        echo "This is message {$i}.<br>";
        echo "FAIL: The message number should be 10.";
    } else {
        echo "PASS: This is message {$i}.";
    }
    echo str_pad('', 5000);
    echo "\r\n\r\n";
    flush();
    usleep(100000);
}
?>

File: LayoutTests/http/tests/multipart/multipart-html.py
#!/usr/bin/env python3

import sys
import time

padding = ' ' * 5000

sys.stdout.buffer.write(
    'Connection: keep-alive\r\n'
    'Content-type: multipart/x-mixed-replace; boundary=boundary\r\n\r\n'
    '--boundary\r\n'
    'Content-Type: text/html\r\n\r\n'
    f'{padding}\n'
    '\n'
    '<script>\n'
    'if (window.testRunner) {\n'
    '    testRunner.dumpAsText();\n'
    '    if (testRunner.setShouldDecideResponsePolicyAfterDelay)\n'
    '        testRunner.setShouldDecideResponsePolicyAfterDelay(true);\n'
    '}\n'
    '</script>\n\n'.encode()
)

for i in range(0, 11):
    sys.stdout.buffer.write(
        b'--boundary\r\n'
        b'Content-Type: text/html\r\n\r\n'
    )
    
    if i < 10:
        sys.stdout.buffer.write(
            'This is message {}.<br>'
            'FAIL: The message number should be 10.'.format(i).encode()
        )
    else:
        sys.stdout.buffer.write('PASS: This is message {}.'.format(i).encode())

    sys.stdout.buffer.write('{}\r\n\r\n'.format(padding).encode())
    sys.stdout.flush()
    time.sleep(0.1)
